diploma-project/services/user.js

```diff
@@ -4,22 +4,26 @@ import {deleteObject, getDownloadURL, listAll, ref, uploadBytesResumable} from "
 import {updateProfile} from "firebase/auth";
 
 export class User {
-    #name = "";
-    #email = "";
-    #photoURL = null;
-    #uid = null;
-    #interests = [];
-    #requests = [];
-    #orders = [];
-
-    constructor(name, email, photoURL, uid, interests, requests, orders) {
+    #name;
+    #email;
+    #photoURL;
+    #uid;
+    #interests;
+    #requests;
+    #craft;
+    #orders;
+    #catalog;
+
+    constructor(name = "", email = "", photoURL = null, uid = null, interests = [], requests = [], craft = null, orders = null, catalog = null) {
```

Comment:
The number of private fields is starting to grow. You may consider using a custom Object that you can pass to the constructor.